### ISSUE TITLE
product angles

### ISSUE THREAD

**Issue by dominictarr** [ssbc/scuttlebot#146]
After my talk yesterday I met several people who had product ideas for ssb.

1. pure replication - just adding reliable replication into their own product. ssb *may* be overkill for this?

2. Auditable - I spoke to one person who had worked for a large engineering company, and there where legal ramifications of emails. One time, they had inspected a building and declared it unsafe, informed someone about it, who later claimed in court not to have received that email. However, he had actually replied to that email, and since email is somewhat decentralized and they had their own copy of the reply they where able to prove that he had infact received the email, and ignored the warning (making him liable)

**Comment by pfraze** [comment]
1. if you dont want a decentralized system, probably yeah.

**Comment by No9** [comment]
Any workflow where non-repudiation is a thing.

I seem to remember talking to someone who thought this would be useful in the creation of legislation?!
Solicitors communications 
Pharmaceutical Production (Facility sign off/maintenance checks/delivery)
Healthcare - Specifically health care records (Actually I might have an idea of a friendly face to talk to)

**Comment by marcelklehr** [comment]
I would be interested in taking [marcelklehr/r-string](http://github.com/marcelklehr/r-string) to the next level by creating an etherpad-like scuttlebot app with your help.

**Comment by pfraze** [comment]
@marcelklehr there's one started at https://github.com/pfraze/textedit.pub, needs updating and some work though. Happy to give you the domain if you want to take it over

**Comment by ahdinosaur** [comment]
overheard from @rdbartlett and is a better way of saying what i think the product angle should be:

> SSB is Heroku for decentralized secure web apps

so the target audience would be web app developers interested in decentralized secure systems but don't want to have to boil the ocean to develop and deploy such an app. this also includes being able to slowly stepwise migrate their existing app(s) to SSB *without negatively affecting the user experience*, which is something i want to make possible for the many apps within [Enspiral](http://enspiral.com).

**Comment by rdbartlett** [comment]
Yeah! :boom: 

I think the "target audience" frame can be a really productive way to have this conversation

**Comment by pfraze** [comment]
@ahdinosaur @rdbartlett yes that's a good angle. Just wrote https://github.com/ssbc/scuttlebot/issues/183#issuecomment-103313044 about deployment

Another angle I've been considering

> A crypto social network, for end-to-end private messaging

SSB distributes public keys and Phoenix uses the social graph to validate identities, so it's a modern take on web-of-trust PKI.

**Comment by ahdinosaur** [comment]
@pfraze the reason i pitch SSB as a tool for app developers instead of an app is that i'm wary of the SSB product being an app that competes with [the app i'm working on](https://github.com/open-app/holodex) and other apps i'm aware of (e.g. i met the founder of [identity.com](http://inflection.com/products/identity) last night who's loosely involved with Enspiral), when i think there's a huge potential to be the underlying infrastructure that allows others to build an ecosystem of social network apps that each focus on a part of the overall problem space and inter-operate through a common protocol.

**Comment by dominictarr** [comment]
We need to have a more detailed discussion on how you would build these
apps on top of ssb.

**Comment by ahdinosaur** [comment]
mm, agreed. my intuition says the best way to figure out how to build apps on top of SSB is to build apps on top of SSB. we can look to how [sandstorm targets app developers](https://github.com/sandstorm-io/sandstorm/wiki/Porting-Guide) as an example.

to throw more spaghetti at the wall, i think the most useful use case of SSB is as a common social data protocol across a wide ecosystem of apps developed and deployed autonomously. maybe i wish SSB was a crypto social private kappa database i could use in my apps. i imagine each database has [delegated access](https://blog.sandstorm.io/news/2015-05-05-delegation-is-the-cornerstone-of-civilization.html) to many feeds, and might defer writes to a remote database.

**Comment by dominictarr** [comment]
@ahdinosaur if only _someone_ wrote another app on top of ssb... :wink:

**Comment by marcelklehr** [comment]
I'm not really familiar with scuttlebot, so: how does it relate to efforts like [GNUnet](https://gnunet.org/)?

**Comment by pfraze** [comment]
Yeah thats a good question, I think there are two things I'd point to. The first is SSB is a relative simple. It does user JSON feeds and (almost) nothing but that. No DHT, no names registry, no consensus algorithm, no proof-of-work. So, think twitter, but with no hosts, and you're free to publish your own json. It's relatively *unfeatured* compared to GNUnet. KISS.

The second is that SSB distributes keypairs, so it's focused toward PKI, and it creates a social network around public keys to identify people. So you look at someone, and you look at who follows them, and you say, "ah, this must be bob, because my friend alice is following him, and alice knows bob." Then you can do things like end-to-end encryption because you've identified the right pubkey. That's the focus.

So the difference is part engineering, part endgoal. The engineering differs in the properties of the network. The endgoal differs entirely - we want to build a log database and a PKI, and GNUnet is more about filesharing and anonymity.

**Comment by ahdinosaur** [comment]
> Be good if ssb was this easy http://zeronet.io/

**Comment by ahdinosaur** [comment]
more spaghetti: ssb as a content sharing network that supports community on top of torrents.
